storage/cloud.py

```diff
@@ -408,6 +415,12 @@ def stream_upload_chunk(self, uuid, offset, length, in_fp, storage_metadata, con
         if bytes_written > 0:
             new_metadata[_CHUNKS_KEY].append(_PartUploadMetadata(chunk_path, offset, bytes_written))
 
+        # Do not leave zero-byte chunks in Object Storage
+        # TODO: Determine a clean way to eliminate calling this function when it's not needed.
+        else:
```

Comment:
Woudn't the above check for zero-length prevent these zero bytes objects to be written in the first place. i.e Don't need to remove them anymore.

Reply:
I don't believe so. In the case where it's explicitly defined as 0 bytes when this file is called, it is short-circuited.

In other cases, I believe that the -1 you explained earlier leads to the creation and writing of a 0 byte file. I'm not exactly sure if that's how they get created, but that initial short-circuit did not fix it for all cases during my personal testing.